pkgs/top-level/all-packages.nix

```diff
@@ -22249,7 +22249,9 @@ with pkgs;
 
   rustc-demangle = callPackage ../development/libraries/rustc-demangle { };
 
-  s2geometry = callPackage ../development/libraries/s2geometry { };
+  s2geometry = callPackage ../development/libraries/s2geometry {
+    abseil-cpp = abseil-cpp_202111;
```

Comment:
Change this to `abseil-cpp = abseil-cpp_202206` and it should build (Works for me at least).